liferay-echo/liferay-portal
# LPS-202724 Disable fields in sitemap UI

## Motivation

[Link to ticket](https://issues.liferay.com/browse/LPS-202724)

The idea behind this PR is to disable the fields in sitemap section based on what the user has selected.

- If they select to use canonical URL, the fields should be disabled
- If they specify nofollow or noindex in robots for any language, the fields should be disabled

## Change summary:

* Adds code to control if the fields in sitemap should be disabled or not


## Test Scenario

* Start the portal
* Go to the Home page, and click on the cog to enter the configuration
* Go to SEO
* Check "Use Canonical URL" -> the sitemap fields will be disabled
* Uncheck "Use Canonical URL" -> the sitemap fields will be enabled
* In robots, write noindex or nofollow -> the sitemap fields will be disabled
* Remove the keywords from robots -> the sitemap fields will be enabled
* Saving the configuration with any of the fields mentioned before and reloading will set the sitemap fields as disabled
